source/frontend/src/features/mapSideBar/research/add/AddResearchContainer.tsx

```diff
@@ -55,12 +67,51 @@ export const AddResearchContainer: React.FunctionComponent<
     (userOverrideCodes: UserOverrideCode[]) => Promise<any | void>
   >('Failed to add Research File');
 
+  // Require user confirmation before adding a property to file
+  // This is the flow for Map Marker -> right-click -> create Research File
   useEffect(() => {
-    if (!!initialForm && !!formikRef.current) {
-      formikRef.current.resetForm();
-      formikRef.current?.setFieldValue('properties', initialForm.properties);
-    }
-  }, [initialForm]);
+    const runAsync = async () => {
+      if (exists(initialForm) && exists(formikRef.current) && needsUserConfirmation) {
+        if (initialForm.properties.length > 0) {
+          const formProperty = initialForm.properties[0];
+          if (formProperty.apiId && (await confirmBeforeAdd(formProperty.apiId))) {
```

Comment:
I have to admit I'm a bit surprised by this implementation. Is there any reason why this was written instead of the standard, api-driven user override pattern we've been using up until now? Perhaps I'm just missing something that precludes the use of that workflow.

Reply:
Also, I think the 
formProperty.apiId
may be incorrect. Since this is an "Add" workflow, the form should not have an apiId yet - that would only be available in update workflows. I think this is why in my testing I was able to save two new research files back-to-back with the same property.

Reply:
let's chat over Teams. Easier than typing here but my interpretation (and we asked this) is that they want the check done "before" saving the file (so we can not do an api-driven user override)

Reply:
> Also, I think the formProperty.apiId may be incorrect. Since this is an "Add" workflow, the form should not have an apiId yet - that would only be available in update workflows. I think this is why in my testing I was able to save two new research files back-to-back with the same property.

The **formProperty.apiId** will indeed be populated if you create a file from an existing property on the map (which has apiId set) and that also belongs to **another** ACQ file. I can show you the flow in the way I tested - maybe I missed something